redisson/src/main/java/org/redisson/codec/FuryCodec.java

```diff
@@ -85,22 +84,35 @@ public FuryCodec(ClassLoader classLoader, boolean registrationRequired, Language
     private final Decoder<Object> decoder = new Decoder<Object>() {
         @Override
         public Object decode(ByteBuf buf, State state) throws IOException {
-            return fury.deserialize(FuryStreamReader.of(new ByteBufInputStream(buf)));
+            MemoryBuffer furyBuffer = MemoryUtils.wrap(buf.nioBuffer());
```

Comment:
nioBuffer can be null. ByteBufInputStream is also zero copy

Reply:
In `FuryStreamReader` implementation, Fury will copy data in `InputStream` into Fury heap buffer for objects whose type is neither primitive array nor bytebuffer. Read byte from `InputStream` is slow, so Fury will read data chunk by chunk.

Reply:
you need to check if nioBuffer() is available using the result of nioBufferCount() method

Reply:
Good catch, updated